dev/release/post-03-website.sh

```diff
@@ -258,9 +258,7 @@ github_url=$(git remote get-url origin | \
                  -e 's,\.git$,,')
 
 echo "Success!"
-echo "1. Open a JIRA issue:"
-echo "  https://issues.apache.org/jira/projects/ARROW/issues/"
```

Comment:
Oh, we don't need JIRA issue for this case?

Reply:
I'm not sure, we directly commit to the asf-branch, which is also weird. We should GHA to render the content of asf-site rather than editing that branch manually.